Makefile

```diff
@@ -0,0 +1,7 @@
+-include .postgres.env
+
+postgres_env:
+	echo "POSTGRES_USER=stringer\nPOSTGRES_PASSWORD=`pwgen -s 20`\nPOSTGRES_HOSTNAME=stringer-postgres\nPOSTGRES_DB=stringer\n" >> .postgres.env
```

Comment:
I don't seem to have `pwgen` installed on my machine. Would it make sense to use `openssl rand` here, too?

Reply:
That was what I had initially, but the `-hex` flag (obviously) only returns hexadecimal characters, and `-base64` has the typical `==` ending that a lot of base64 strings have. It also seemed like I couldn't control the exact number of characters like with `pwgen`. I can look into a few more options.

Reply:
So the issue with `-hex` is the limited character set? I'm not finding any clear message as to whether that is a problem. Looking at Heroku database urls, it looks like they're using hex characters on their end, so that's a vote of confidence. (They seem to prefix it with a `p` in some cases.)

Reply:
Well, given it's a password, restricting the candidate character set makes it a lot easier to guess.

Reply:
You made me go and learn about password strength. According to [Wikipedia](https://en.wikipedia.org/wiki/Password_strength):

> Due to currently understood limitations from fundamental physics, there is no expectation that any digital computer (or combination) will be capable of breaking 256-bit encryption via a brute-force attack.

Which may be the reason that 64 digit hex passwords are used, since 16^64 == 2^256, which achieves the highest recommended password entropy. That comes out to 115792089237316195423570985008687907853269984665640564039457584007913129639936 possible variations.

Reply:
Hey, looking at Keepass for quality is actually a great idea. Here's 20 hex characters: 
![image](https://github.com/stringer-rss/stringer/assets/18634426/2ce84342-4ec9-4a8a-b45e-cccfffd98aa2)

I tried a few times and it ranged from 85-100 bits of entropy. According to [this chart](https://i.imgur.com/gfYw57t.png) that's still ~3 million years at the fastest attempts/s to crack it, on average, so I think I agree with you that it's probably enough for a default. Now to figure out how to make it yield 20 characters, though :D

Reply:
Hey, looking at Keepass for quality is actually a great idea. Here's 20 hex characters:

![image](https://github.com/stringer-rss/stringer/assets/18634426/0fb90d3b-6918-4bdc-9d6e-36f2a19f8270)

I tried a few times and it ranged from 70-75 bits of entropy. According to [this chart](https://i.imgur.com/gfYw57t.png) that's 4.5 months at the fastest attempts/s to crack it, and half as long on average, so I think we'd have to make it longer still. 25 characters is 100 bits of entropy, so that should be enough (~200 million years to crack on average).

(NB: originally I had posted with an image that had non-hex characters and reached a different conclusion)

Reply:
Heroku uses 64 characters for the db password, so I think we could probably run with that. `openssl rand -hex 32` generates a 64 character string.